account_invoice_inter_company/models/account_move.py

```diff
@@ -197,6 +197,24 @@ def _prepare_invoice_data(self, dest_company):
         )
         return vals
 
+    def button_draft(self):
+        for move in self:
+            company = move._find_company_from_invoice_partner()
+            if company and not move.auto_generated:
+                inter_invoice_posted = self.sudo().search(
+                    [("auto_invoice_id", "=", move.id), ("state", "=", "posted")]
+                )
+                if inter_invoice_posted:
+                    raise UserError(
+                        _(
+                            "You can't modify this invoice as it has an inter company "
+                            "invoice that's in posted state.\n"
+                            "Invoice %s to %s"
+                        )
+                        % (company.name, inter_invoice_posted.name)
```

Comment:
The order of the parameters is switched and you don't need `company`, just `inter_invoice_posted.partner_id.display_name`.

Reply:
sure :+1: